### ISSUE TITLE
Task with name 'connectedUnspecifiedAndroidTest' not found

### ISSUE THREAD

**Issue by mateuszkwiecinski** [Karumi/Shot#207]
### Expected behaviour
I want to apply the plugin and see tasks mentioned in the documentation.

### Actual behaviour
build fails at configuration phase with following exception:
```
Caused by: org.gradle.api.UnknownTaskException: Task with name 'connectedUnspecifiedAndroidTest' not found in project ':example-activity-alias'.
        at org.gradle.api.internal.tasks.DefaultTaskCollection.createNotFoundException(DefaultTaskCollection.java:102)
        at org.gradle.api.internal.tasks.DefaultTaskCollection.createNotFoundException(DefaultTaskCollection.java:46)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.getByName(DefaultNamedDomainObjectCollection.java:333)
        at org.gradle.api.internal.tasks.DefaultTaskCollection.getByName(DefaultTaskCollection.java:46)
        at com.karumi.shot.ShotPlugin.addTasksFor(ShotPlugin.scala:193)
        at com.karumi.shot.ShotPlugin.addTaskToVariant(ShotPlugin.scala:106)
        at com.karumi.shot.ShotPlugin.$anonfun$addTasksToAppModule$1(ShotPlugin.scala:88)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext$CurrentApplication$1.execute(DefaultUserCodeApplicationContext.java:100)
```
Build scan: https://scans.gradle.com/s/fgqo2isc6hhhu

### Steps to reproduce
Apply the plugin lazily, i.e. using
```
pluginManager.withPlugin("com.android.application") {
        pluginManager.apply("shot")
```
it seems to be working when applied "normally" at the root of each project.

Looking at the source code, it seems like the plugin does not support configuration avoidance and tries to evaluate project properties eagerly. 
i.e.
https://github.com/Karumi/Shot/blob/ef7721cf43c516b1a5c291182294ce98bb47b5fb/shot/src/main/scala/com/karumi/shot/ShotPlugin.scala#L87

AFAIU every `.all` should be replaced with `configureEach` call https://blog.gradle.org/preview-avoiding-task-configuration-time

But that ☝️ is just a guess there might be some other reasons. 

If repro project is needed: I tried to replace facebook's library with `Shot` here: https://github.com/usefulness/easylauncher-gradle-plugin/blob/31f31317aa112d7a5f9a6a1d0d4023a76b55dd98/sample/build.gradle#L54

### Version of the library
[`Fix storage usage for compose only users with emulator API <= 28`](https://github.com/Karumi/Shot/releases/tag/5.10.3)

**Comment by mateuszkwiecinski** [comment]
🤦 
I was terribly wrong! the `unspecified` comes from from one of my `buildTypes` 🤦 
```
buildTypes {
        register("unspecified") {
        }
```

So yeah, again, basically the issue happens at `getByName()` call, where a `named()` should be used instead

https://github.com/Karumi/Shot/blob/ef7721cf43c516b1a5c291182294ce98bb47b5fb/shot/src/main/scala/com/karumi/shot/ShotPlugin.scala#L193

this should allow to create tasks for `debug` buildType, which I understand is the only buildType supported right now 👀 

So tl;dr I faced: https://github.com/Karumi/Shot/issues/202 which combined with lack of configuration avoidance results in unclear error message. 

I should update the first secion:

### Expected behaviour
I can apply the plugin to a project with custom buildTypes

**Comment by tikitu** [comment]
Yup, we ran into this too: `Task with name 'connectedDevelopQaAndroidTest' not found in project ':app'.`. Same reason: a custom `buildType`. I'm very far from confident in this area, but I wonder if it would be correct to only create tasks for the build type configured in `testBuildType` (with `debug` as the default)?

**Comment by pedrovgs** [comment]
Wow a lot of information here!! Thanks for reporting @mateuszkwiecinski 😃  I'm taking a look.

**Comment by mateuszkwiecinski** [comment]
I can confirm the plugin forks for me now 👌 Thank you @pedrovgs 🚀
